src/header.tsx

```diff
@@ -62,13 +66,19 @@ export const FilesCardHeader = ({ currentFilter, onFilterChange, isGrid, setIsGr
     );
 };
 
-const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
-    const [isOpen, setIsOpen] = useState(false);
-    const onToggleClick = isOpen => setIsOpen(!isOpen);
-    const onSelect = (ev, itemId) => {
-        setIsOpen(false);
-        setSortBy(itemId);
-        localStorage.setItem("files:sort", itemId);
+const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }:
+    { isGrid: boolean, setIsGrid: React.Dispatch<React.SetStateAction<boolean>>,
+      sortBy: string, setSortBy: React.Dispatch<React.SetStateAction<string>>}) => {
+    const [isOpen, setIsOpen] = useState<boolean>(false);
+    const onToggleClick = (isOpen: boolean) => setIsOpen(!isOpen);
+    const onSelect = (_ev?: React.MouseEvent, itemId?: string | number) => {
+        if (typeof itemId === "string") {
+            setIsOpen(false);
+            setSortBy(itemId);
+            localStorage.setItem("files:sort", itemId);
+        } else {
+            console.error("unexpected itemId type", itemId);
```

Comment:
Let's use https://github.com/cockpit-project/cockpit/issues/20298 . No need to block this PR on it, so that we don't block the TS investigations. But this should be a follow-up.

Reply:
It now uses cockpit.assert